Add documentation with DocFX

J-M-White opened the pull request:
In response to #5, I followed the [Bonsai Docfx Tutorial](https://bonsai-rx.org/docs/articles/documentation-docfx.html) to add documentation that is consistent with other Bonsai packages. The information is the same as what was previously in the readme, with a plan to add more in future. 

The website has not been published to Github Pages yet.

J-M-White committed:
Add minimal Docfx documentation

Untrack files in .gitignore

Standard property formatting

Delete tutorials, extra images, year in app footer, references to other packages.

Move operator descriptions to separate articles.

banchan86 commented:
hey @J-M-White just wanted to drop a note to say I will be reviewing this soon!

banchan86 commented on .github/workflows/docs.yml:
```diff
@@ -0,0 +1,49 @@
+        uses: microsoft/setup-msbuild@v2
+        
+      - name: Restore NuGet Packages
+        run: msbuild -t:restore src\Bonsai.ZeroMQ.sln
```
There are references to another package here that need to be modified, but we are actually revamping the CI/CD workflow for testing/building packages/publishing documentation. but we can leave the infrastructure revamp for another PR as we are still finalizing it.

banchan86 reviewed (commented):
Good job getting the docfx website up! I wrote some comments that could be addressed in my review. One general high level comment - as the bonsai workflow containers are not being used in this package (thus far) some related assets are redundant could be removed (such as the .bonsai environment, build.ps1 and etc). There are some areas where they could be useful (for instance, either the `TriggerRecording` or saving data from SpikeFetch/SpikeStream operators if that is possible).

J-M-White commented:
Great! Thanks, @banchan86. I'll go through and address these comments.

J-M-White commented:
@glopesdev, thanks for that insight! Exactly like you said, what `SpikeGLXInput` is doing right now is laying groundwork and then calling `cs_sglx_fetch` repeatedly. Technically, when it is being triggered by an input it calls the API function `cs_sglx_fetchLatest`; however, that function itself calls `cs_sglx_fetch`. I think renaming `SpikeGLXInput` to `SglxFetch` would make a lot of sense.

In a similar vein, do you think `TriggerRecording` should be renamed to `TriggerGT` or `SglxTriggerGT` to align with the name of the underlying API function? `DigitalOutput` may be a more interesting case - it looks like the newest version of the API actually replaces the `c_sglx_setDigitalOut` function I've used with `c_sglx_ni_DO_set` for NI devices, and adds `c_sglx_obx_AO_set` for OneBoxes. Maybe it's worth renaming it now to something that explicitly indicates it's only for NI devices, so that there's no confusion if a separate operator is later added to interface with Oneboxes. What do you think?

Any review you have time to do (on naming or anything else) is much appreciated!

banchan86 commented on docs/articles/spikeglx-triggerrecording.md:
```diff
@@ -0,0 +1,17 @@
+
+For example, the below workflow starts a new gate immediately, and then toggles the trigger every five seconds. This results in SpikeGLX saving five-second periods of the run every ten seconds. 
+:::workflow
+![TriggerRecording](../workflows/GettingStarted-TriggerRecording.bonsai)
```
Ah the new bonsai workflow  is a great inclusion and does help make things clearer! I just noticed the name for the first GT node has no spaces `(G,T)=(1,-1)` but the others have spaces `(G,T) = (-1,1)`.

banchan86 commented on README.md:
```diff
@@ -1,42 +1,14 @@
-# Bonsai.SpikeGLX
-[Bonsai](http://bonsai-rx.org/) package for interfacing with [SpikeGLX](https://billkarsh.github.io/SpikeGLX/).
+# Bonsai - SpikeGLX
+**Bonsai.SpikeGLX** is a [Bonsai](https://bonsai-rx.org/) package for interfacing with the [SpikeGLX](https://billkarsh.github.io/SpikeGLX/) recording software for [Neuropixels](https://www.neuropixels.org/) probes. 
```
So we have been a bit inconsistent with it in the past, but I think we prefer to highlight package names as inline code using backticks (`) (Making a mental note for myself to add it to the style guide). Only making a suggested change for this one instance but there are other instances if you want to adopt this change.

```suggestion
`Bonsai.SpikeGLX` is a [Bonsai](https://bonsai-rx.org/) package for interfacing with the [SpikeGLX](https://billkarsh.github.io/SpikeGLX/) recording software for [Neuropixels](https://www.neuropixels.org/) probes.

J-M-White replied:
This should now be fixed across all instances.

banchan86 commented on docs/articles/getting-started.md:
```diff
@@ -0,0 +1,33 @@
+
+A brief summary of the functionality of each of these operators is provided here. A more detailed description of each may be found on the associated Reference page.
+> [!NOTE]
+> All **Bonsai.SpikeGLX** operators require SpikeGLX's remote command server to be running. A full description of how to do this can be found in the [SpikeGLX User Manual](https://github.com/billkarsh/SpikeGLX/blob/master/Markdown/UserManual.md#remote-command-servers). Make sure to note the `Host` (IP Address) and `Port` the command server is listening on. These are properties that must be set fo ...
```
Any reason why the user manual points to the GitHub file instead of the website? Is it not updated?

```suggestion
> All `Bonsai.SpikeGLX` operators require SpikeGLX's remote command server to be running. A full description of how to do this can be found in the [SpikeGLX User Manual](https://billkarsh.github.io/SpikeGLX/Sgl_help/UserManual.html#remote-command-servers). Make sure to note the `Host` (IP Address) and `Port` the command server is listening on. These are properties that must be set for each `Bonsai.SpikeGLX` operator.

banchan86 reviewed (commented):
Hi @J-M-White the documentation changes look great, especially with the creation of the individual operator articles and workflow examples. Just added a few comments that are docs specific if you want to address them for now. I can take another look after @glopesdev has updated the package repository and reviewed the naming/structure!

J-M-White committed:
Update docs/apidoc/Bonsai_SpikeGLX_TriggerRecording.md

Co-authored-by: Shawn Tan <banchan@gmail.com>

J-M-White committed:
Add consistent naming in Trigger-Recording example workflow

J-M-White committed:
Use backticks (`) to highlight all package names.

J-M-White committed:
Update link to the remote command server page in the SpikeGLX manual.

J-M-White committed:
Improve documentation

J-M-White committed:
Improve documentation

J-M-White committed:
Merge branch '5-add-documentation-with-docfx' of https://github.com/FeeLab/bonsai-spikeglx into 5-add-documentation-with-docfx

J-M-White committed:
Add example workflow for DigitalOutput

glopesdev committed:
Slight formatting changes for consistency

The shortening of the first subsection name is to help with URL
normalization and looks better in TOC.

glopesdev committed:
Add package README file

This is what will be displayed specifically under the NuGet website
(not on the Bonsai package manager).

The base template text was used, which is essentially a one-sentence
description and a list of relevant links to install, docs, license, and
repo.

glopesdev reviewed (commented):
This looks great, thank you so much for being so thorough!

I found it easier to add review comments to a separate PR to your branch (https://github.com/FeeLab/bonsai-spikeglx/pull/1), happy to discuss there or here.

glopesdev committed:
Remove unused actions

J-M-White committed:
Merge pull request #1 from glopesdev/documentation-review

Documentation review.
- Small formatting changes in docs README and Getting Started article
- Add package README (for NuGet website)
- Remove unused actions.

J-M-White commented:
The proposed changes in the separate PR all looked good to me! It's now been merged. 

I'm glad to see that the **official** Bonsai policy is to use semicolons for separating long items in a list (I'm a big fan of this but have met with some resistance in the past).

glopesdev reviewed (commented):
@J-M-White thanks, I think we're almost there! Just noticed two files (`logo.svg` and `favicon.ico`) which are not needed anymore as they are included in the submodule. Sorry I didn't spot them in my earlier PR!

Other than that everything looks great and we should be ready to release. One last question I had is what version number to use for the release. Is it okay to use our usual starting point at 0.1.0? Or did you happen to use this one internally before?

P.S.: re. bullet list punctuation I am actually not too picky either way as long as there is consistency. In general we try to follow the [MSDN Writing Style Guide](https://learn.microsoft.com/en-us/style-guide/welcome/) for alignment with .NET documentation.

Interestingly, I just checked and they do recommend to avoid semicolons and conjunctions entirely in their [Punctuation](https://learn.microsoft.com/en-us/style-guide/scannable-content/lists#punctuation) guide for lists. I'm used to seeing them in formal writing all the time, but there does seem to be mild consensus on avoiding them for web browsing as they are considered "too complex", so I'm torn here.

J-M-White committed:
Delete favicon.ico and logo.svg

J-M-White commented:
@glopesdev, releasing as version 0.2.0 sounds good!

glopesdev commented on docs/.gitignore:
This file also not needed since the site builds in the `artifacts` folder now.

glopesdev reviewed (commented):
@J-M-White Initially spotted another unused file, so decided to do a full review and other things came up. All minor and should be quick to fix, I expect to merge after this round, thanks for your patience!

J-M-White commented on docs/workflows/GettingStarted-Trigger.bonsai:
```diff
@@ -0,0 +1,158 @@
+        <Text><![CDATA[]]></Text>
+      </Expression>
+      <Expression xsi:type="GroupWorkflow">
+        <Name>(G,T)=(1,-1)</Name>
```
Yeah, I see what you mean. When I originally set it up this way I had imagined that someone might want to algorithmically determine `Gate` and `Trigger` values in different circumstances and then pass those to a single `Trigger` operator. However, that leads to it being more unwealdy in situations like the example.

J-M-White committed:
Clean up documentation
Delete unecessary `.gitignore`s, remove exclusion of `C_Sglx_namespace` from `docs/filter.yml`, change "Reference" section to "API" in `docs/toc.yml`.

banchan86 commented on docs/toc.yml:
```diff
@@ -1,2 +1,4 @@
-- name: API
+- name: Manual
+  href: articles/
+- name: Reference
```
@glopesdev took note of all the documentation comments in this review and will roll them into my refresh of the documentation (https://github.com/bonsai-rx/docs/issues/107) soon!
